### ISSUE TITLE
feat: make the hint the rightmost longpress key when the keyboard is a right-to-left keyboard

### ISSUE THREAD

**Issue by rmlockwood** [keymanapp/keyman#10916]
### Is your feature request related to a problem? Please describe.

It's great that the default hint is what is on the longpress keys, but having the default be the leftmost one of the longpress keys is opposite of what the user would want. The user will longpress and then scan or move right to left through the options.

### Describe the solution you'd like

When the keyboard is for a right-to-left language, use the rightmost longpress key. When there are multiple (like more than 3) the bottom rightmost is the one I would want to be the default hint. This rightmost key is the one I'm judging to be the most frequently used and the closes distance for the user.

### Keyman apps

- [X] Keyman for Android
- [X] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

16

**Comment by mcdurdin** [comment]
Thanks for the feedback! I've put this into the Future bucket for now, as we already have a lot on the boil for 18.0, but will review when we are doing issue triage.

**Comment by jahorton** [comment]
There are some complications to consider, though.

![image](https://github.com/keymanapp/keyman/assets/25213402/00973c19-d9b4-4630-bf3f-822a948b5b71)

There are cases with some of our keyboards where there are two rows of longpress keys.  Would it be safe to assume you'd prefer the rightmost _on the first row_ to be the default?

----

An even better question, in my mind:  would it be better if the longpress keys _flowed_ from right to left, instead of left to right as they currently do?  Our current implementations are obviously LTR-biased, which is what's given rise to this problem.

In this case, if we know that the keyboard is for an RTL language...

![image](https://github.com/keymanapp/keyman/assets/25213402/33e8bc00-992f-4899-a3f8-1aa65d58031f)

we could have the first longpress key display at the top-right in _both_ the touch-layout editor _and_ in the live subkey menu... which would have the net result of the first subkey being at the top-right for RTL keyboards while remaining top-left for LTR keyboards.

Our current subkey default:
- Does any subkey for the base key claim "default"?  If so, _that_.
- Does the current base key have a subkey with the same key ID?
- Use the first subkey.

This discussion assumes you're relying on that last entry - using the "first" subkey.  It's just... the system currently doesn't account for RTL perspectives when determining "first".

**Comment by rmlockwood** [comment]
Yes, the rightmost being the default would be great for a RTL language keyboard. Having it flow right-to-left would be great.

________________________________
